test/unit/client_test.coffee

```diff
@@ -107,6 +108,12 @@ describe "Client", ->
         error = _err
       return error
 
+    it "returns promise and resolves it", ->
+      promise = client.push(exception)
```

Comment:
This illustrates the use case for this, correct? They want to know when the ID is available so they can do their thing with the notice ID? If yes, would it not have been enough to accept a callback and call that rather than adding an impl of Promise?